aws_lambda_powertools/utilities/feature_flags/feature_flags.py

```diff
@@ -259,11 +277,22 @@ def get_enabled_features(self, *, context: Optional[Dict[str, Any]] = None) -> L
         for name, feature in features.items():
             rules = feature.get(schema.RULES_KEY, {})
             feature_default_value = feature.get(schema.FEATURE_DEFAULT_VAL_KEY)
+            boolean_feature = feature.get(
+                schema.FEATURE_DEFAULT_VAL_TYPE_KEY, True
+            )  # backwards compatability ,assume feature flag
+            if not boolean_feature:
+                self.logger.debug(f"skipping feature because it is not a boolean feature flag, name={name}")
+                continue
```

Comment:
```suggestion
```

Just found this piece when fixing Mypy and adding GA notices and I'm removing it -- Let me know if my reading is incorrect.

In the docs, we call out `get_all_enabled_features` will return [any features that match the input context](https://awslabs.github.io/aws-lambda-powertools-python/latest/utilities/feature_flags/#getting-all-enabled-features) - This code block however ignores non-boolean and doesn't honour our docs, additional tests failed too.

Looking up the code to confirm whether the docs was accidentally incorrect, docstrings confirm it:

```python
def get_enabled_features(self, *, context: Optional[Dict[str, Any]] = None) -> List[str]:
		...
		
        Returns
        ----------
        List[str]
            list of all feature names that either matches context or have True as default
```